src/wallet/coinselection.h

```diff
@@ -197,6 +197,32 @@ struct OutputGroup
  */
 [[nodiscard]] CAmount GetSelectionWaste(const std::set<CInputCoin>& inputs, CAmount change_cost, CAmount target, bool use_effective_value = true);
 
+struct SelectionResult
+{
+    /** Set of inputs selected by the algorithm to use in the transaction */
+    std::set<CInputCoin> m_selected_inputs;
+    /** The target the algorithm selected for. Note that this may not be equal to the recipient amount as it can include non-input fees */
+    const CAmount m_target;
+    /** Whether the input values for calculations should be the effective value (true) or normal value (false) */
+    bool m_use_effective{false};
+
+    explicit SelectionResult(const CAmount target)
+        : m_target(target) {}
```

Comment:
delete default ctor, to make it impossible to create a `SelectionResult` with no specified target?

Reply:
Done